cpp/src/arrow/json/reader_test.cc

```diff
@@ -35,10 +35,11 @@ using util::string_view;
 
 using internal::checked_cast;
 
-class ReaderTest : public ::testing::TestWithParam<bool> {
+class ReaderTest : public ::testing::TestWithParam<std::pair<bool, bool>> {
```

Comment:
This will start being a bit cryptic. Perhaps:

```suggestion
struct ReaderTestParam {
  bool use_threads;
  bool use_readahead;
};

class ReaderTest : public ::testing::TestWithParam<ReaderTestParam> {
```

Reply:
Done